src/profile.ts

```diff
@@ -0,0 +1,132 @@
+import * as path from 'path';
+import * as fs from 'fs';
+
+import * as jwt from 'jsonwebtoken';
+
+import debugModule from 'debug';
+const debug = debugModule('everyauth:profile');
+
+const JWT_CACHE_MARGIN = 1000 * 60 * 5; // 5 minute margin before refreshing the key.
+const settingsName = 'settings.json';
+const keyDir = 'keys';
+const publicKeyFileName = 'pub';
+const privateKeyFileName = 'pri';
+const jwtAlgorithm = 'RS256';
+
+interface IProfilePki {
+  algorithm: 'RS256';
+  audience: string;
+  issuer: string;
+  subject: string;
+  kid: string;
+  privateKey: string;
+  publicKey: string;
+}
+
+export interface IProfile {
+  created: string;
+  updated: string;
+  account: string;
+  subscription: string;
+  baseUrl: string;
+  issuer: string;
+  subject: string;
+  keyPair: string;
+  kid: string;
+  pki: IProfilePki;
+
+  token?: string;
+}
+
+export let cachedFoundProfile: IProfile;
+
+export const cachedJwt: {
+  expiresAt: number;
+  token: string;
+} = { expiresAt: 0, token: '' };
+
+export const loadProfile = async (profileName?: string): Promise<IProfile> => {
+  if (cachedFoundProfile) {
+    return cachedFoundProfile;
+  }
+
+  // Look for the EVERYAUTH_PROFILE_JSON object
+  if (process.env.EVERYAUTH_PROFILE_JSON) {
```

Comment:
I prefer not to reference env vars like this; I would instead use a configuration utility that loads these values, or specify it via function parameter.